CONTRIBUTING.rst

```diff
@@ -388,26 +388,70 @@ They should run on small mocked data, cover a representative range of parameters
       For more information about this coding approach,
       see `test-driven development <https://en.wikipedia.org/wiki/Test-driven_development>`_.
 
+We use `pytest <https://docs.pytest.org/en/6.2.x/contents.html>`_ to run our tests.
+
+If you are not familiar with pytest,
+have a look at this `introductory video <https://www.youtube.com/watch?v=mzlH8lp4ISA>`_
+by one of the pytest core developer.
+
+In general tests for a specific module (say `nilearn/image/image.py`)
+are kept in a `tests` folder in separate module
+with a name that matches the module being tested
+(so in this case `nilearn/image/tests/test_image.py`).
+
+When you have added a test you can check your changes worked
+and didn't break anything run `pytest nilearn`.
+To do quicker checks it's possible to run only a subset of tests:
+
+.. code-block:: bash
+
+      pytest -v nilearn/module/tests/test_module.py
+
+Fixtures
+########
+
+If you need to some special "set up" for your tests
+(for example you need to generate some data, or a NiftiImage object or a file...)
+you can use `pytest fixtures <https://docs.pytest.org/en/6.2.x/fixture.html>`_
+to help you mock this data
+(more information on pytest fixtures in `this video <https://www.youtube.com/watch?v=ScEQRKwUePI>`_).
+
+Fixture are recognizable because they have a `@pytest.fixture` decorator.
+Fixtures that are shared by many tests modules can be found in `nilearn/conftest.py`
+but some fixures specific to certain modules can also be kept in that testing module.
+
+Seending
+########
+
 Tests must be seeded to avoid random failures.
 For objects using random seeds (e.g. scikit-learn estimators), pass either
-a  `np.random.RandomState` or an `int` as the seed.
-When your test use random numbers,  those must be generated through:
+a `np.random.RandomState` or an `int` as the seed.
```

Comment:
Maybe for a separate PR but shouldn't we update to use the Generator API as mentioned in https://github.com/nilearn/nilearn/issues/3331#issuecomment-1684047671?

Reply:
I updated it here already: what do you think?

Reply:
Yes, good with me! At some point we can update our tests to use it but that really should be in a separate PR

Reply:
agreed

I will take care of it soon but it should be done in a separate or several PRs